### ISSUE TITLE
Build problem.

### ISSUE THREAD

**Issue by ABrilliantBeast** [intel-go/nff-go#259]
Hitting an error at top level make.  Smells like a GOROOT problem, but I am using a standard Ubuntu 16.04 install.  One difference seems to be the Ubuntu install installs to /usr/lib/go-1.9 ran than /usr/local.  Spinning up a new VM using the standard location but having issues with our cloud.  Seen this before?

> make[2]: Entering directory '/mnt/Code/gocode/src/github.com/intel-go/nff-go/test/framework'
Checking for AVX support... AVX and AVX2
go generate
stringer: checking package: types.go:8:2: could not import encoding/json (can't find import: )
types.go:12: running "stringer": exit status 1
Makefile:11: recipe for target 'apptype_string.go' failed
make[2]: *** [apptype_string.go] Error 1
make[2]: Leaving directory '/mnt/Code/gocode/src/github.com/intel-go/nff-go/test/framework'
../mk/intermediate.mk:13: recipe for target 'framework' failed
make[1]: *** [framework] Error 2
make[1]: Leaving directory '/mnt/Code/gocode/src/github.com/intel-go/nff-go/test'
mk/intermediate.mk:13: recipe for target 'test' failed
make: *** [test] Error 2

go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/mnt/Code/gocode"
GORACE=""
GOROOT="/usr/lib/go-1.9"
GOTOOLDIR="/usr/lib/go-1.9/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build721947677=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"

go version
go version go1.9.4 linux/amd64

**Comment by gshimansky** [comment]
You did not install a `stringer` package which is necessary to build test framework. Execute `go get  golang.org/x/tools/cmd/stringer`. To install all NFF-Go dependencies at once you can execute `scripts/get-depends.sh` script.

**Comment by ABrilliantBeast** [comment]
I had done that and it is there.  I went back to go version 1.9.1 and it builds now.  Seems like I remember so monkeying around with generate in one of the golang 1.9.X drops  1.9.4 and 1.10 are broke.

**Comment by gshimansky** [comment]
This is strange, because on develop branch 1.10 works for my just fine.

**Comment by ABrilliantBeast** [comment]
I was just using the master branch, but I switched over to develop and saw the same problem.  I have tried with go 1.9.1, 1.9.4, and 1.10.1.  It has only completed on 1.9.1.   Here is the error

>$ git branch
> \* (HEAD detached at origin/develop)
  master
$ make -j8
Checking for AVX support... AVX and AVX2
make -C nff-go-base 
make[1]: Entering directory '/mnt/Code/gocode/src/github.com/intel-go/nff-go/nff-go-base'
Checking for AVX support... AVX and AVX2
make[1]: Nothing to be done for 'all'.
\<snip>
make[3]: Entering directory '/mnt/Code/gocode/src/github.com/intel-go/nff-go/test/stability/testResend'
Checking for AVX support... AVX and AVX2
go build testResend.go 
stringer: checking package: types.go:8:2: could not import encoding/json (can't find import: )
types.go:12: running "stringer": exit status 1
Makefile:11: recipe for target 'apptype_string.go' failed
make[2]: *** [apptype_string.go] Error 1
make[2]: Leaving directory '/mnt/Code/gocode/src/github.com/intel-go/nff-go/test/framework'
../mk/intermediate.mk:13: recipe for target 'framework' failed
make[1]: *** [framework] Error 2
make[1]: *** Waiting for unfinished jobs....
go build rw_example.go

**Comment by gshimansky** [comment]
I see now that the root of the problem is that stringer cannot find encoding/json:

`stringer: checking package: types.go:8:2: could not import encoding/json (can't find import: )`

Are you sure that you use correct runtime version (pointed to by GOROOT) when you switch Go versions? 

I didn't use standard Ubuntu Go version because it is usually outdated, I always downloaded a binary from golang.org.

**Comment by ABrilliantBeast** [comment]
My original machine is still at 1.9.4

I created a new VM from scratch 16.04.3 and just downloaded the golang tarball and installed to /usr/local

Golang 1.10.1 fails the same way, then switched to 1.9.1, which worked.  I'm going through the golang releases now to see where it breaks. 1.9.2 works.

**Comment by ABrilliantBeast** [comment]
OK, so I suddenly couldn't reproduce this problem on the new machine, using the golang tarball install rather than an Ubuntu package, eventhough I swear I saw it.  Maybe I had a memory lapse over lunch.  I went to the extend of clearing cache, deleting everthing under GOPATH and building from scratch and it all worked.  I then removed to Ubuntu package from the original machine and installed the tarball.  It now compiles too.  It must be some issue with the Ubuntu package.  Thanks.
